### PR TITLE
[CI/CD] Add implementation for `test-dbt-installation-source` workflow

### PR DESCRIPTION
**Pull request by alexander-smolyakov**
**Description**:
This PR adds implementation for the `test-dbt-installation-source` workflow to test dbt adapters [installation from sources](https://docs.getdbt.com/docs/get-started/source-install).

_Changelog_:
- Added initial implementation;
- Updated workflow docs;

**Related issue**:
- #62 

**Run examples**:
- https://github.com/dbt-labs/actions/actions/runs/3967169576;
- https://github.com/dbt-labs/actions/actions/runs/3967208445;

### CODE REVIEW

File: .github/workflows/test-dbt-installation-source.yml [hunk 1]
```diff
@@ -32,10 +38,55 @@ jobs:
   source-integration-test:
     runs-on: ubuntu-latest
 
+    strategy:
+      fail-fast: false
+      matrix:
+        branch: ["1.1.latest", "1.2.latest", "1.3.latest"]
```
Comment (emmyoop) [inline .github/workflows/test-dbt-installation-source.yml:44]: Same question here with determining the matrix dynamically so we don't have to manually update this with each release.  i've made an attempt to do the same (although not currently working) in another PR - https://github.com/dbt-labs/dbt-core/pull/6682

File: .github/workflows/test-dbt-installation-source.yml [hunk 2]
```diff
@@ -32,10 +38,55 @@ jobs:
   source-integration-test:
```
Comment (emmyoop) [inline .github/workflows/test-dbt-installation-source.yml:38]:
```suggestion
  source-installation-test:
```

**Review by emmyoop — CHANGES_REQUESTED** [review]

File: .github/workflows/test-dbt-installation-source.yml (hunk 2 above)
Comment (alexander-smolyakov) [reply .github/workflows/test-dbt-installation-source.yml]: Fixed https://github.com/dbt-labs/actions/pull/68/commits/65e27042ba1233977a0b327bfd92204afbf772e6

File: .github/workflows/test-dbt-installation-source.yml (hunk 1 above)
Comment (alexander-smolyakov) [reply .github/workflows/test-dbt-installation-source.yml]: I added a job to fetch metadata about protected branches via GitHub API: https://github.com/dbt-labs/actions/pull/68/files#diff-40a4b2214e076aa5da41caf0b207a245fb1699b11aa4c72f667c541be035a030R41-R80

Also, I updated the main job to use a dynamic matrix.

Run examples:
- https://github.com/dbt-labs/actions/actions/runs/3986791048;
- https://github.com/dbt-labs/actions/actions/runs/3986773070;

File: .github/workflows/test-dbt-installation-main.yml [hunk 3]
```diff
@@ -13,14 +13,29 @@ permissions:
 
 on:
   workflow_dispatch:
+    inputs:
+      test-installation-method:
+        description: "Choice installation method to test"
```
Comment (emmyoop) [inline .github/workflows/test-dbt-installation-main.yml:18]:
```suggestion
        description: "Choose installation method(s) to test"
```

**Review by emmyoop — APPROVED** [review]
Another small description change.  Otherwise good to merge.

File: .github/workflows/test-dbt-installation-main.yml (hunk 3 above)
Comment (alexander-smolyakov) [reply .github/workflows/test-dbt-installation-main.yml]: Fixed in https://github.com/dbt-labs/actions/pull/68/commits/df4eb6faaedf3bad5d6354e0c464f4c7a5864b63

**Pull request merged**